"No registration found for username: ..." when case does not match

Hi again.

Once again, first of all, thanks so much for this project. Push Notifications might seem like a minor thing, but once you've lived without them, it feels weird to not have them.

I was troubleshooting a certificate renewal issue on one of my 2 mail systems that run this, and noticed that xapsd seems to only match usernames in a case sensitive fashion.
One of my users set up their account starting the username with a capital letter, but on their devices they have signed in with their username all lowercase (does iOS mail just automatically lower-case the username that is entered?). Obviously this works for logging in and getting/sending mail via IMAP/SMTP, but xapsd reports a warning every time a message is delivered to their inbox:

`time="2023-03-03T15:24:27+11:00" level=warning msg="No registration found for username: Username"`

When I look in the json database, their username is recorded lowercase. In testing with the user, they do not receive push notifications (though they had not realised they should be so not the end of the world)

Just wondering if case sensitivity is intentional or if an update could be implemented to fix this?

I'd offer to make a PR myself (if it isn't an intentional feature), but I have no experience in Go and would need 3 management levels and corporate legal approval to make or contribute to open source projects. Happy to throw beer money around for it though 😅

Ok, I'm probably missing something starkly obvious but when I try to build it, I get:

`/root/go/pkg/mod/github.com/spf13/afero@v1.9.3/internal/common/adapters.go:16:8: package io/fs is not in GOROOT (/usr/src/io/fs)`

io/fs was introduced in Go 1.16, and I still had 1.15 installed so I updated golang to golang-1.16, but no change, and then updated it fully to 1.19, but still the same message, and no completed build.
This is on Debian 11, kernel 5.10.0-14-amd64, in case that helps.

--- a/internal/socket.go
+++ b/internal/socket.go
@@ -7,6 +7,7 @@ import (
 	"github.com/julienschmidt/httprouter"
 	log "github.com/sirupsen/logrus"
 	"net/http"
+	"strings"
 )
 
 type httpHandler struct {
@@ -92,7 +93,7 @@ func (httpHandler *httpHandler) handleRegister(writer http.ResponseWriter, reque
 	}
 
 	// Register this email/account-id/device-token combination
-	err = httpHandler.db.AddRegistration(reg.Username, reg.ApsAccountId, reg.ApsDeviceToken, reg.Mailboxes)
+	err = httpHandler.db.AddRegistration(strings.ToLower(reg.Username), reg.ApsAccountId, reg.ApsDeviceToken, reg.Mailboxes)
 	if err != nil {
 		log.Errorf("Failed to register client:: %s", err)
 		writer.WriteHeader(http.StatusInternalServerError)
@@ -129,6 +130,11 @@ func (httpHandler *httpHandler) handleNotify(writer http.ResponseWriter, request
 		writer.WriteHeader(http.StatusBadRequest)
 		return
 	}
+
+	// Dovecot supports case sensitive usernames, but postfix doesn't
+	// So we only care about lowercase users for now
+	// This isn't an exploit either, since we don't send any email contents via push
+	notify.Username = strings.ToLower(notify.Username)
 
 	isMessageNew := false
 	// check if this is an event for a new message

